test/filters/test_experimental_voxel_grid.cpp

```diff
@@ -0,0 +1,383 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Point CLoud Library (PCL) - www.pointclouds.org
+ * Copyright (c) 2021-, Open Perception
+ *
+ * All rights reserved
+ */
+
+#include <pcl/filters/experimental/voxel_grid.h>
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/test/gtest.h>
+#include <pcl/point_types.h>
+
+#include <cmath>
+#include <limits>
+
+using namespace pcl;
+using namespace pcl::io;
+
+PointCloud<PointXYZ>::Ptr cloud(new PointCloud<PointXYZ>);
+PointCloud<PointXYZRGB>::Ptr cloud_rgb(new PointCloud<PointXYZRGB>);
+PointCloud<PointXYZRGBA>::Ptr cloud_rgba(new PointCloud<PointXYZRGBA>);
+
+const float PRECISION = Eigen::NumTraits<float>::dummy_precision() * 2;
+
+template <typename PointT>
+void
+EXPECT_POINT_EQ(const PointT& pt1, const PointT& pt2)
+{
+  const auto& pt1_vec = pt1.getVector4fMap();
+  const auto& pt_vec = pt2.getVector4fMap();
+  EXPECT_TRUE(pt1_vec.isApprox(pt_vec, PRECISION))
+      << "Point1: " << pt1_vec.transpose() << "\nPoint2: " << pt_vec.transpose()
+      << "\nnorm diff: " << (pt1_vec - pt_vec).norm() << std::endl;
+}
+
+template <typename>
+void
+EXPECT_POINT_EQ(const PointXYZRGB& pt1, const PointXYZRGB& pt2)
```

Comment:
Is the point type unknown that we have to use template specialization?

Reply:
https://github.com/tin1254/pcl/blob/8c893c030d94200803774e820fe8e3caedc06072/test/filters/test_experimental_voxel_grid.cpp#L74-L82

Is there any better way than using template specialization? because I don't wanna to duplicate the sorting

Reply:
I couldn't think of any. LGTM